yt/frontends/stream/io.py

```diff
@@ -110,11 +110,10 @@ def _read_particle_coords(self, chunks, ptf):
                 )
 
     def _read_smoothing_length(self, chunks, ptf, ptype):
-        for data_file in sorted(
+        data_file = sorted(
```

Comment:
I removed the loop here because it looks superfluous

Reply:
Hmmmmmm.  I believe you that it's likely true, but looking at it I don't see how.